client/state/global-sidebar/selectors.ts

```diff
@@ -14,14 +14,24 @@ export const getShouldShowGlobalSidebar = (
 	_: AppState,
 	siteId: number,
 	sectionGroup: string,
-	sectionName: string // eslint-disable-line @typescript-eslint/no-unused-vars
+	sectionName: string
 ) => {
-	return (
+	if (
 		sectionGroup === 'me' ||
 		sectionGroup === 'reader' ||
-		sectionGroup === 'sites-dashboard' ||
 		( sectionGroup === 'sites' && ! siteId )
-	);
+	) {
+		return true;
+	}
+	// Show the global sidebar for the sites dashboard
+	// If on hosting section, also check if the feature flag is enabled
+	if (
+		sectionGroup === 'sites-dashboard' &&
+		( sectionName !== 'hosting' || isEnabled( 'layout/dotcom-nav-redesign-v2' ) )
+	) {
+		return true;
+	}
+	return false;
```

Comment:
I understand the behavior for `sectionGroup === 'sites-dashboard'` as follows:

- When no site is selected (`/sites`), we display the global sidebar in its expanded form.
- When a site is selected, on any page being viewed (e.g., the hosting configuration page or the GitHub deployment page), we display the global sidebar in its collapsed form.

If so, could we update this entire condition in `getShouldShowGlobalSidebar` to the following? (sorry if I'm wrong)

```ts
	return (
		sectionGroup === 'me' ||
		sectionGroup === 'reader' ||
		( isEnabled( 'layout/dotcom-nav-redesign-v2' )
			? sectionGroup === 'sites-dashboard' && ! siteId
			: sectionGroup === 'sites-dashboard' ) || 
		( sectionGroup === 'sites' && ! siteId )
	);
```

The collapsed global sidebar would be displayed appropriately, as covered in `getShouldShowCollapsedGlobalSidebar` https://github.com/Automattic/wp-calypso/blob/add/hosting-config/client/state/global-sidebar/selectors.ts#L37-L47